exercises/complex-numbers/complex_numbers_test.rb

```diff
@@ -0,0 +1,207 @@
+require 'minitest/autorun'
+require_relative 'complex_numbers'
+
+# Common test data version: 1.0.0 '117d062'
+class ComplexNumbersTest < Minitest::Test
+  def test_imaginary_unit
+    # #skip
+    expected = ComplexNumber.new([-1, 0])
+    assert_equal expected, ComplexNumber.new([0, 1]) * ComplexNumber.new([0, 1])
+  end
+
+  def test_add_purely_real_numbers
+    #skip
+    expected = ComplexNumber.new([3, 0])
+    assert_equal expected, ComplexNumber.new([1, 0]) + ComplexNumber.new([2, 0])
+  end
+
+  def test_add_purely_imaginary_numbers
+    #skip
+    expected = ComplexNumber.new([0, 3])
+    assert_equal expected, ComplexNumber.new([0, 1]) + ComplexNumber.new([0, 2])
+  end
+
+  def test_add_numbers_with_real_and_imaginary_part
+    #skip
+    expected = ComplexNumber.new([4, 6])
+    assert_equal expected, ComplexNumber.new([1, 2]) + ComplexNumber.new([3, 4])
+  end
+
+  def test_subtract_purely_real_numbers
+    #skip
+    expected = ComplexNumber.new([-1, 0])
+    assert_equal expected, ComplexNumber.new([1, 0]) - ComplexNumber.new([2, 0])
+  end
+
+  def test_subtract_purely_imaginary_numbers
+    #skip
+    expected = ComplexNumber.new([0, -1])
+    assert_equal expected, ComplexNumber.new([0, 1]) - ComplexNumber.new([0, 2])
+  end
+
+  def test_subtract_numbers_with_real_and_imaginary_part
+    #skip
+    expected = ComplexNumber.new([-2, -2])
+    assert_equal expected, ComplexNumber.new([1, 2]) - ComplexNumber.new([3, 4])
+  end
+
+  def test_multiply_purely_real_numbers
+    #skip
+    expected = ComplexNumber.new([2, 0])
+    assert_equal expected, ComplexNumber.new([1, 0]) * ComplexNumber.new([2, 0])
+  end
+
+  def test_multiply_purely_imaginary_numbers
+    #skip
+    expected = ComplexNumber.new([-2, 0])
+    assert_equal expected, ComplexNumber.new([0, 1]) * ComplexNumber.new([0, 2])
+  end
+
+  def test_multiply_numbers_with_real_and_imaginary_part
+    #skip
+    expected = ComplexNumber.new([-5, 10])
+    assert_equal expected, ComplexNumber.new([1, 2]) * ComplexNumber.new([3, 4])
+  end
+
+  def test_divide_purely_real_numbers
+    #skip
+    expected = ComplexNumber.new([0.5, 0])
+    assert_equal expected, ComplexNumber.new([1, 0]) / ComplexNumber.new([2, 0])
+  end
+
+  def test_divide_purely_imaginary_numbers
+    #skip
+    expected = ComplexNumber.new([0.5, 0])
+    assert_equal expected, ComplexNumber.new([0, 1]) / ComplexNumber.new([0, 2])
+  end
+
+  def test_divide_numbers_with_real_and_imaginary_part
+    #skip
+    expected = ComplexNumber.new([0.44, 0.08])
+    assert_equal expected, ComplexNumber.new([1, 2]) / ComplexNumber.new([3, 4])
+  end
+
+  def test_absolute_value_of_a_positive_purely_real_number
+    #skip
+    expected = ComplexNumber.new([5])
```

Comment:
This is the first usage of a single element array as a constructor argument.
Is that a good thing to do?

Reply:
You are right.